data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/StreamRecordConverterTest.java

```diff
@@ -476,6 +480,39 @@ void remove_record_without_use_old_image_on_delete_uses_new_image() throws Excep
         verify(bytesProcessedSummary).record(record.dynamodb().sizeBytes());
     }
 
+    @Test
+    void test_writeSingleRecordToBuffer_with_userIdentity() throws Exception {
+
+        when(streamConfig.getStreamViewForRemoves()).thenReturn(StreamViewType.OLD_IMAGE);
+
+        final Identity userIdentity = Identity.builder()
+                .principalId("dynamodb.amazonaws.com")
```

Comment:
Please include a test case where the userIdentity is not null, but the principal is something other than this known principle.

It would cover the case of `&& "dynamodb.amazonaws.com".equals(record.userIdentity().principalId())`.